### PR TITLE
B fix elbv2 cookie name set on wrong type

### PR DESCRIPTION
**Pull request by timothyjrogers**
Relates #26735

Output from acceptance testing: I did not exercise acceptance testing.

### CODE REVIEW

File: internal/service/elbv2/target_group.go [hunk 1]
```diff
@@ -1037,7 +1037,9 @@ func flattenTargetGroupStickiness(attributes []*elbv2.TargetGroupAttribute) ([]i
 				m["cookie_duration"] = duration
 			}
 		case "stickiness.app_cookie.cookie_name":
-			m["cookie_name"] = aws.StringValue(attr.Value)
+			if sType, ok := m["type"].(string); !ok || sType == "app_cookie" {
```
Comment (everops-miked) [inline internal/service/elbv2/target_group.go:1040]: this looks good but where is sType being set? are we sure that the iteration will happen in the right order?

**Comment by everops-miked** [comment]
Maybe add a test case that will exercise this ?

File: internal/service/elbv2/target_group.go (hunk 1 above)
Comment (timothyjrogers) [reply internal/service/elbv2/target_group.go:1040]: If I'm understanding this code correctly the stickiness attributes passed into `flattenTargetGroupStickiness` are read from the live Target Group resource via the API, so at the point they are being flattened all necessary entries are guaranteed to be in the array and are just being reformatted into a map.

**Comment by timothyjrogers** [comment]
> Maybe add a test case that will exercise this ?

I'm thinking through the test case now and I'm trying to decide how to set it up as its not clear how the cookie name would become set on a target group using `lb_cookie`. In your scenario from [#26735 ](https://github.com/hashicorp/terraform-provider-aws/issues/26735) was this target group previously using `app_cookie` and updated at some point to use `lb_cookie`, or has it always used `lb_cookie`?

**Comment by everops-miked** [comment]
I think the app cookie was set and unset manually in the console at some point.

**Pull request closed without merging**